src/core/qgsmaplayer.cpp

```diff
@@ -94,6 +94,43 @@ QgsMapLayer::~QgsMapLayer()
   delete mStyleManager;
 }
 
+void QgsMapLayer::clone( QgsMapLayer *layer, bool deep ) const
+{
+  layer->setBlendMode( blendMode() );
+
+  QMap<QString, QgsMapLayerStyle> styles = styleManager()->mapLayerStyles();
+  for ( QString s : styles.keys() )
+  {
+    layer->styleManager()->addStyle( s, styles.value( s ) );
+  }
+
+  layer->setName( name() );
+  layer->setShortName( shortName() );
+  layer->setMinimumScale( minimumScale() );
+  layer->setMaximumScale( maximumScale() );
+  layer->setScaleBasedVisibility( hasScaleBasedVisibility() );
+  layer->setTitle( title() );
+  layer->setAbstract( abstract() );
+  layer->setKeywordList( keywordList() );
+  layer->setDataUrl( dataUrl() );
+  layer->setDataUrlFormat( dataUrlFormat() );
+  layer->setAttribution( attribution() );
+  layer->setAttributionUrl( attributionUrl() );
+  layer->setMetadataUrl( metadataUrl() );
+  layer->setMetadataUrlType( metadataUrlType() );
+  layer->setMetadataUrlFormat( metadataUrlFormat() );
+  layer->setLegendUrl( legendUrl() );
+  layer->setLegendUrlFormat( legendUrlFormat() );
+  layer->setDependencies( dependencies() );
+  layer->setCrs( crs() );
+  layer->setCustomProperties( mCustomProperties );
+
+  if ( deep )
```

Comment:
is this syill required? or can we drop the `deep` param, and the id is always unique.